homeassistant/components/waterfurnace.py

```diff
@@ -110,8 +112,11 @@ def shutdown(event):
             try:
                 self.data = self.client.read()
 
-            except ConnectionError:
-                # attempt to log back in if there was a session expiration.
+            except wf.WFException:
+                # WFExceptions are things the WF library understands
+                # that pretty much can all be solved by logging in and
+                # back out again.
+                _LOGGER.exception("Failed to read data, attempting to recover")
                 try:
                     self.client.login()
                 except Exception:  # pylint: disable=broad-except
```

Comment:
It's not allow from component to make a broad exception. Please change this

Reply:
ok, though that's already in the code and been in the release. I can trim that in a follow up, but it shouldn't prevent fixing other issues.

Reply:
This is a quite special case, used for shutting down a dedicated thread cleanly. I am not sure that the broad `except` is so bad in this case where the alternative is an ugly log?

Reply:
Yeah, I see it's his own thread, so it's okay